core/src/main/java/com/netease/arctic/io/ArcticHadoopFileIO.java

```diff
@@ -212,17 +208,13 @@ public Iterable<FileInfo> listPrefix(String prefix) {
   @Override
   public void deletePrefix(String prefix) {
     tableMetaStore.doAs(() -> {
-      Path toDelete = new Path(prefix);
-      FileSystem fs = getFs(toDelete);
+      Path prefixToDelete = new Path(prefix);
+      FileSystem fs = getFs(prefixToDelete);
       try {
-        if (!fs.delete(toDelete, true)) {
-          throw new IOException("Fail to delete directory:" + prefix + " recursively, " +
-              "file system return false, need to check the hdfs path");
-        }
+        return fs.delete(prefixToDelete, true /* recursive */);
```

Comment:
It appears that in Hadoop 3, `FileSystem#delete` no longer returns a boolean value. However, removing the return value check from Hadoop 2 may result in the failure of the delete operation being ignored.